common/src/main/java/org/apache/comet/vector/CometDictionary.java

```diff
@@ -30,7 +30,7 @@ public class CometDictionary implements AutoCloseable {
   private final int numValues;
 
   /** Decoded dictionary values. We only need to copy values for decimal type. */
-  private ByteArrayWrapper[] binaries;
+  private volatile ByteArrayWrapper[] binaries;
```

Comment:
This is not directly related but forgot to add based on https://github.com/apache/datafusion-comet/pull/705#discussion_r1688697138